setuptools_rust/build.py

```diff
@@ -349,10 +351,42 @@ def get_dylib_ext_path(self, ext, target_fname):
         assert modpath not in build_ext.ext_map
         build_ext.ext_map[modpath] = ext
         try:
-            return build_ext.get_ext_fullpath(target_fname)
+            ext_path = build_ext.get_ext_fullpath(target_fname)
         finally:
             del build_ext.ext_map[modpath]
 
+        if '.abi3.' in ext_path:
+            return ext_path
+        # Examples: linux_x86_64, linux_i686, manylinux2014_aarch64, manylinux_2_24_armv7l
+        plat_name = self.plat_name.lower().replace('-', '_').replace('.', '_')
+        if not plat_name.startswith(('linux', 'manylinux')):
+            return ext_path
+
+        arch_parts = []
+        arch_found = False
+        for item in plat_name.split('_'):
+            if item.startswith(('linux', 'manylinux')):
+                continue
+            if item.isdigit() and not arch_found:
+                # manylinux_2_24_armv7l arch should be armv7l
+                continue
+            arch_found = True
+            arch_parts.append(item)
+        target_arch = '_'.join(arch_parts)
+        host_platform = sysconfig.get_platform()
+        host_arch = host_platform.rsplit('-', 1)[1]
+        # Remove incorrect platform tag if we are cross compiling
+        if target_arch and host_arch != target_arch:
+            ext_dir = os.path.dirname(ext_path)
+            ext_name = os.path.basename(ext_path)
+            # rust.cpython-38-x86_64-linux-gnu.so to (rust.cpython-38-x86_64-linux-gnu, .so)
+            ext_name, ext_ext = os.path.splitext(ext_name)
+            # rust.cpython-38-x86_64-linux-gnu to (rust, .cpython-38-x86_64-linux-gnu)
+            ext_name, _plat_tag = os.path.splitext(ext_name)
```

Comment:
We do a similar pattern about line ~310, maybe can refactor to share a common util function?

Reply:
Still would like to see this, maybe call `strip_platform_and_extension()` ?